src/ProductView/ProductView.js

```diff
@@ -121,12 +161,30 @@ class ProductView extends Component {
             this.setState({filter: updatedFilters});
         }
     }
-    
+
     setActiveEventType(eventType) {
         this.fetchEventsFor(eventType);
         this.setState({activeEventType: eventType});
     }
-    
+
+    setProductConfiguration(configurationId) {
+        this.setState({
+            loading: true
+        });
+        if(configurationId) {
+            this.setState({
+                showAllConfigurations: false
+            });
+            this.configurationId = configurationId;
+            this.fetchConfiguration();
+        }
+        else {
+            this.setState({
```

Comment:
Eventually overwriting setState. Consider executing setState at the very end of the function.